scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TFRecordSpec.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.tensorflow
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.tensorflow.FeatureKind._
+import com.spotify.scio.values.SCollection
+import io.circe.generic.auto._
+import io.circe.syntax._
+import org.apache.beam.sdk.io.Compression
+
+/** Contains metadata about TFRecords. Useful to read the records later on. */
+trait TFRecordSpec {
+  val LATEST_VERSION: Int = 1
+  def toJson(sc: ScioContext, compression: Compression): SCollection[String]
```

Comment:
Don't seen the reason for `sc: ScioContext, compression: Compression` to be part of `toJson`? Shouldn't both `sc` and `compression` be params for specific impl of `TFRecordSpec`. In case of `compression` if it's common it could be a declared in the trait.